File: ueueo-framework/ueueo-authorization/src/main/java/com/ueueo/authorization/permissions/UserPermissionValueProvider.cs
using System.Linq;
using System.Threading.Tasks;
using Volo.Abp.Security.Claims;

namespace Volo.Abp.Authorization.Permissions;

public class UserPermissionValueProvider : PermissionValueProvider
{
    public const String ProviderName = "U";

    @Override
    public String Name => ProviderName;

    public UserPermissionValueProvider(IPermissionStore permissionStore)
        : base(permissionStore)
    {

    }

    @Override
    public PermissionGrantResult CheckAsync(PermissionValueCheckContext context)
    {
        var userId = context.Principal?.FindFirst(AbpClaimTypes.UserId)?.Value;

        if (userId == null)
        {
            return PermissionGrantResult.Undefined;
        }

        return PermissionStore.IsGrantedAsync(context.Permission.Name, Name, userId)
            ? PermissionGrantResult.Granted
            : PermissionGrantResult.Undefined;
    }

    @Override
    public MultiplePermissionGrantResult CheckAsync(PermissionValuesCheckContext context)
    {
        var permissionNames = context.Permissions.Select(x => x.Name).Distinct().ToArray();
        Check.NotNullOrEmpty(permissionNames, nameof(permissionNames));

        var userId = context.Principal?.FindFirst(AbpClaimTypes.UserId)?.Value;
        if (userId == null)
        {
            return new MultiplePermissionGrantResult(permissionNames);
        }

        return PermissionStore.IsGrantedAsync(permissionNames, Name, userId);
    }
}

File: ueueo-framework/ueueo-authorization/src/main/java/com/ueueo/authorization/permissions/UserPermissionValueProvider.java
package com.ueueo.authorization.permissions;

import com.ueueo.ID;
import org.apache.commons.collections4.CollectionUtils;
import org.springframework.util.Assert;

import java.util.List;
import java.util.stream.Collectors;

public class UserPermissionValueProvider extends PermissionValueProvider {
    public final String ProviderName = "U";

    @Override
    public String getName() {
        return ProviderName;
    }

    public UserPermissionValueProvider(IPermissionStore permissionStore) {
        super(permissionStore);

    }

    @Override
    public PermissionGrantResult check(PermissionValueCheckContext context) {
        ID userId = context.getPrincipal().findUserId();

        if (userId == null) {
            return PermissionGrantResult.Undefined;
        }

        return permissionStore.isGranted(context.getPermission().getName(), getName(), userId.toString())
                ? PermissionGrantResult.Granted
                : PermissionGrantResult.Undefined;
    }

    @Override
    public MultiplePermissionGrantResult check(PermissionValuesCheckContext context) {
        List<String> permissionNames = context.getPermissions().stream().map(PermissionDefinition::getName).distinct().collect(Collectors.toList());
        Assert.isTrue(CollectionUtils.isNotEmpty(permissionNames), "PermissionNames must not empty!");

        ID userId = context.getPrincipal().findUserId();
        if (userId == null) {
            return new MultiplePermissionGrantResult(permissionNames);
        }

        return permissionStore.isGranted(permissionNames, getName(), userId.toString());
    }
}
